pkg/services/ngalert/database.go

```diff
@@ -0,0 +1,120 @@
+package ngalert
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+func getAlertDefinitionByID(alertDefinitionID int64, sess *sqlstore.DBSession) (*AlertDefinition, error) {
+	alertDefinition := AlertDefinition{}
+	has, err := sess.ID(alertDefinitionID).Get(&alertDefinition)
+	if !has {
+		return nil, ErrAlertDefinitionNotFound
+	}
+	if err != nil {
+		return nil, err
+	}
+	return &alertDefinition, nil
+}
+
+// DeleteAlertDefinitionByID is a handler for deleting an alert definition.
+// It returns models.ErrAlertDefinitionNotFound if no alert definition is found for the provided ID.
+func (ng *AlertNG) DeleteAlertDefinitionByID(query *DeleteAlertDefinitionByIDQuery) error {
```

Comment:
Doesn't have to be public:
```suggestion
// deleteAlertDefinitionByID is a handler for deleting an alert definition.
// It returns models.ErrAlertDefinitionNotFound if no alert definition is found for the provided ID.
func (ng *AlertNG) deleteAlertDefinitionByID(query *DeleteAlertDefinitionByIDQuery) error {
```